controllers/UIController/index.js

```diff
@@ -236,27 +245,49 @@ export default class UIController<Props, State>
     }
 
     // Events
+    getBottomInsetAdjustment() {
+        let adjustment;
+        if (Platform.OS === 'android') {
+            adjustment = UIController.onGetAndroidDisplayCutout().top / PixelRatio.get();
+        } else if (Platform.OS === 'ios') {
+            adjustment = -this.getSafeAreaInsets().bottom;
+        }
+        return adjustment;
+    }
+
+    adjustBottomInset(
+        container: NativeMethodsMixinType,
+        keyboardFrame: $ReadOnly<{ screenY: number, height: number }>,
+        animation: ?AnimationParameters,
+        readjustTimeout: number,
+    ) {
+        container.measure((relX, relY, w, h, x, y) => {
+            const keyboardOverlapHeight = Math.max(
+                (y + h) - keyboardFrame.screenY,
+                keyboardFrame.height,
+            );
+            const bottomInset = keyboardOverlapHeight + this.getBottomInsetAdjustment();
+            this.setBottomInset(Math.max(0, bottomInset), animation);
+            if (Platform.OS === 'android' && (readjustTimeout > 0)) {
+                setTimeout(() => {
+                    this.adjustBottomInset(container, keyboardFrame, animation, 0);
+                }, readjustTimeout);
+            }
+        });
+    }
+
     onKeyboardWillShow(event: KeyboardEvent) {
-        const end = event.endCoordinates;
+        const keyboardFrame = event.endCoordinates;
         const animation = UIController.getKeyboardAnimation(event);
         const { container } = this;
-        if (!(container && container.measure)) {
-            this.setBottomInset(end.height, animation);
-            return;
+        if (!!container && !!container.measure) {
+            this.adjustBottomInset(container, keyboardFrame, animation, 300);
```

Comment:
I suggest to leave a comment for `300`, OR move this constant above and provide some proper method.